File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/PhotoResponse.java
package example.develop.davidoh.java_android_mvp_example.data;

import com.google.gson.annotations.SerializedName;

public class PhotoResponse {
    @SerializedName("photos")
    private PhotoPageInfo photoPageInfo;

    private int code;
    private String message;
    private String stat;

    public PhotoPageInfo getPhotoPageInfo() {
        return photoPageInfo;
    }

    public void setPhotoPageInfo(PhotoPageInfo photoPageInfo) {
        this.photoPageInfo = photoPageInfo;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getStat() {
        return stat;
    }
}

File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/PhotoResponse.kt
package example.develop.davidoh.java_android_mvp_example.data

data class PhotoResponse(
    val photos: Photos,
    val stat: String,
    val code: Int,  // error code
    val message: String
)

data class Photos(
    val page: Int,
    val pages: String,
    val perpage: Int,
    val photo: List<Photo>,
    val total: String
)

data class Photo(
    val farm: Int,
    val id: String,
    val isfamily: Int,
    val isfriend: Int,
    val ispublic: Int,
    val owner: String,
    val secret: String,
    val server: String,
    val title: String
) {
    fun getImageUrl() = "https://farm$farm.staticflickr.com/$server/${id}_$secret.jpg"
}

